middleware.ml

```diff
@@ -78,13 +85,32 @@ let redirect_to_dashboard reqd ?(msg = "") () =
   Httpaf.Reqd.respond_with_string reqd response msg;
   Lwt.return_unit
 
-let cookie_value_from_auth_cookie cookie =
+let http_response ~title ?(header_list = []) ?(data = "") reqd http_status =
+  let code = Httpaf.Status.to_code http_status
+  and success = Httpaf.Status.is_successful http_status in
+  let status = { Utils.Status.code; title; data; success } in
+  let data = Utils.Status.to_json status in
+  let headers =
+    Httpaf.Headers.(
+      add_list
+        (of_list
+           [
+             ("Content-Type", "application/json");
+             ("Content-length", string_of_int (String.length data));
+           ])
+        header_list)
+  in
+  let response = Httpaf.Response.create ~headers http_status in
+  Httpaf.Reqd.respond_with_string reqd response data;
+  Lwt.return_unit
+
+let cookie_value_from_cookie cookie =
   match String.split_on_char '=' (String.trim cookie) with
   | _ :: s :: _ -> Ok (String.trim s)
```

Comment:
Why are there all these `String.trim` around? Does cookie values have lots of whitespaces that have to be ignored?

Reply:
(see above, maybe best to leave this for a future)